source/crud/tests/unified/estimatedDocumentCount.yml

```diff
@@ -0,0 +1,213 @@
+                  }
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount errors correctly on 4.9.0 or greater"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ aggregate ]
+              closeConnection: true
+      - name: estimatedDocumentCount
+        object: *collection0
+        expectError: { isError: true }
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: &pipeline
+                  - $collStats: { count: {} }
+                  - $group: {
+                    _id: 1,
+                    n: { $sum: $count },
+                  }
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount uses count on less than 4.9.0"
+    runOnRequirements:
+      - maxServerVersion: "4.8.99"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                count: *collection0Name
+              commandName: count
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with maxTimeMS on less than 4.9.0"
+    runOnRequirements:
+      - maxServerVersion: "4.8.99"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          maxTimeMS: 6000
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                count: *collection0Name
+                maxTimeMS: 6000
+              commandName: count
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount on non-existent collection on less than 4.9.0"
+    runOnRequirements:
+      - maxServerVersion: "4.8.99"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection1
+        expectResult: 0
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                count: *collection1Name
+              commandName: count
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount errors correctly on less than 4.9.0"
+    runOnRequirements:
+      - minServerVersion: "4.0.0"
+        maxServerVersion: "4.8.99"
+        topologies: [ single, replicaset ]
+      - minServerVersion: "4.2.0"
+        maxServerVersion: "4.8.99"
+        topologies: [ sharded ]
```

Comment:
`failCommand` isn't available until 4.0 for replset/standalone, and 4.2 for sharded, so I have to set `minServerVersion`s for this test.